### PR TITLE
Fix the problem that documentation for methods and properties is not shown #5881

### PR DESCRIPTION
**Pull request by junichi11**
- https://github.com/apache/netbeans/issues/5881
- Get also the prefix until the caret position to calculate the correct anchor

### Example
```php
class GH5881 {
    /**
     * Property $prop.
     */
    public $prop;
    /**
     * Property $prop_aa.
     */
    public $prop_aa;
    /**
     * Property $prop_aa_bbb.
     */
    public $prop_aa_bbb;
    /**
     * Property $prop_aa_bb_cc.
     */
    public $prop_aa_bb_cc;

    /**
     * Method method().
     *
     * @return void
     */
    public function method(): void {
    }

    /**
     * Method method_aa().
     *
     * @return void
     */
    public function method_aa(): void {
    }

    /**
     * Method method_aa_bbb().
     *
     * @return void
     */
    public function method_aa_bbb(): void {
    }

    /**
     * Method method_aa_bb_cc().
     *
     * @return void
     */
    public function method_aa_bb_cc(): void {
    }
}
$test = new GH5881();
$prop1 = $test->prop_aa_bbb;
$prop2 = $test->prop_aa_bb_cc;
$test->method_aa_bbb();
$test->method_aa_bb_cc();
```
nb-php-gh-5881-cc-doc-regression-property

nb-php-gh-5881-cc-doc-regression-method

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java [hunk 1]
```diff
@@ -335,7 +335,9 @@ public CodeCompletionResult complete(CodeCompletionContext completionContext) {
         // GH-4494 if query type is documentation, get a whole identifier as a prefix
         boolean upToOffset = queryType != QueryType.DOCUMENTATION;
         prefix = getPrefix(info, caretOffset, upToOffset, PrefixBreaker.WITH_NS_PARTS);
+        String prefixUntilCaret = getPrefix(info, caretOffset, true, PrefixBreaker.WITH_NS_PARTS); // GH-5881
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java:338]: If `upToOffset` is `true`, we can skip this call and use `prefix` directly, no?

**Review by tmysik — APPROVED** [review]
Thanks for the fix and all the tests.

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java (hunk 1 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java:338]: Yes, You are right! Will fix it. Thanks.
```java
String prefixUntilCaret = upToOffset ? prefix : getPrefix(info, caretOffset, true, PrefixBreaker.WITH_NS_PARTS); // GH-5881
```

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java [hunk 2]
```diff
@@ -358,10 +360,11 @@ public CodeCompletionResult complete(CodeCompletionContext completionContext) {
         }
         request.extraPrefix = searchPrefix;
 
+        int prefixLengthForAnchor = queryType == QueryType.DOCUMENTATION ? prefixUntilCaret.length() : prefix.length();
         request.anchor = caretOffset
                 // can't just use 'prefix.getLength()' here cos it might have been calculated with
                 // the 'upToOffset' flag set to false
-                - prefix.length();
+                - prefixLengthForAnchor;
```
Comment (junichi11) [inline php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java:367]: Then
```suggestion
                - prefixUntilCaret.length();
```

**Comment by junichi11** [comment]
@tmysik Thank you for your review and your time :)

File: php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java (hunk 1 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java]: Fixed: https://github.com/apache/netbeans/compare/1fcbaa35a3af943c7db9a4ad40fe9d4531f56a69..946bd31030a69df973f41e94f458af61c38a9ac5

**Comment by tmysik** [comment]
@junichi11 Sure, no problem!

**Pull request merged**